src/SafeLog.jl

```diff
@@ -0,0 +1,48 @@
+module SafeLog
+
+const SAFE_LOG = Ref(false)
+
+"""
+    set_safe_log(x::Bool)
+
+Turn on/off SAFE_LOG, to plot histograms without/with negative values. By default is `false`.
+"""
+function set_safe_log(x::Bool)
+    SAFE_LOG[]=x
+end
+
+
+"""
+    _clip_for_log(c_vec)
+
+Internal fuction to clip counts of histogram. 
+"""
+function _clip_counts!(c_vec)
+    min_positive = eps()
+    @. c_vec = max(c_vec, min_positive)
+end
+
+
+"""
+    _clip_counts!(c_vec,el_vec, eh_vec)
+
+Internal fuction to clip counts and errors of histogram. 
+"""
+function _clip_counts!(c_vec, el_vec, eh_vec)
+    
+    # Set the clipping, and make copy of starting counts
+    min_positive = eps()
+    c_vec_def = copy(c_vec)
+
+    # Clip bin counts
+    @. c_vec = max(c_vec, min_positive)
+
+    # clip lower errors
+    mask =  c_vec - el_vec .< min_positive
+    el_vec[mask] = c_vec[mask] .- min_positive
+
+    # clip higher errors
+    @. eh_vec = max(eh_vec - (c_vec - c_vec_def), min_positive)
```

Comment:
what is the logic of this?

Reply:
The logic here is that if you camp bin counts, you need to adjust also the upper error accordingly. E.g. your data point is y = - 0.1  σ = 0.2. If you clip the central value to y' = 0, the upper error should be set to σ' = σ - (y' - y) =  0.1. Now if central value is very low or error is very small, you could get  σ' < 0, and therefore you take the max between that and `min_positive` value. Does that make sense?

Reply:
All this is in the spirit that even if you're clamping stuff to look at log scales, it could still be interesting information that there's a data point right below the 0, but with an error big enough that makes it compatible with a positive value, and up to which positive value the error extends.